.github/workflows/step_test.yml

```diff
@@ -0,0 +1,42 @@
+on:
+  workflow_call:
+    secrets:
+      CODECOV_TOKEN:
+        required: false
+        description: Codecov token
+
+permissions:
+  contents: read
+
+jobs:
+  pre-commit:
+    name: Run pre-commit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+      - uses: pre-commit/action@v3.0.0
+
+  checks:
+    name:
+      Check 🐍 ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```

Comment:
Related to comment about doing tests in hatch environments. Python version matrix can be also declared there, making this redundant.

Reply:
I think it is better to have github action handle python versions on this one. It is consistently maintained and has development versions as well. As for the `hatch` environment will address in the other comment

Reply:
The python interpreters would be the same regardless. Hatch envs matrix also allows to run the tests on multiple versions locally (or somewhere), which can be Python that is actually packaged in Fedora, rather than gh ubuntu.

Reply:
> The python interpreters would be the same regardless. Hatch envs matrix also allows to run the tests on multiple versions locally (or somewhere), which can be Python that is actually packaged in Fedora, rather than gh ubuntu.

I am not sure how this works, so I would deffer this to you to show an example later on ok? The github packages tend to take the most recent patch updates as they come and also include the development versions before they are available in downstream (will add `3.13` as experimental soon). If hatch can do something similar and has good reporting, it could be beneficial. I was thinking of how to add the mypy test matrix as well. So the main things that should be equivalent:
- Support for the most recent python patches as well as development python versions
- Support for marking a whole environment as experimental where the failing tests are reported but do not fail the CI
- Good reporting of which tests come from which environment

Let's check this one out in a later PR where we can compare them ok?

Reply:
I can see
```
[[tool.hatch.envs.test.matrix]]
python = ["3.9", "3.11", "3.12"]
```
in pyproject.toml already. You could do the same for let's say cov env used here. In github workflow, it would be just a matter of installing those python interpreters to be available.

Reply:
This one, I think we can drop all `CI.yml` tests and move them to a more `tmt` native test-suite. Maybe have only the `sphinx-builder` tests.  But on the topic of `hatch`, is the stdout report usable enough that you can navigate which python version it came from? If so we could consolidate them if we want to keep the GH action workflows